CONTRIBUTING.md

```diff
@@ -175,3 +175,90 @@ For testing and development you can find self-signed certificates under
 ```bash
  openssl req -nodes -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -subj '/CN=localhost'
 ```
+
+## Logging Principles
+Logging plays a crucial role for administrators, users and developers of our tool, which is why it
+is important to have certain standards in regards on when and how to use which type of logger.
+
+The following sections will outline
+- what information should be logged.
+- when information should be logged.
+- how information should be logged.
+
+### What information should be logged.
+#### Unknown-Error Log Message
+These types of messages are things such as an unexpected exception was thrown, which should
+provider relative information and context on the exception.
+The following questions should be answered with these log messages.
+- What error happened
+- what data was used to cause this error
+- when did the error happen
+- ...
+
+
+#### Informational Log Message
+These types of log messages are used to give application administrators contexts to user errors,
+users errors can be for example, a stream could not be created due to incorrect user data.
+The following questions should be answered with these log messages.
+- when did the error occur
+- what data was used to cause the error
+- what api path did the user call
+- ...
+
+#### Exclude sensitive information
+Be careful when working on parts of the application that could internationally contain sensitive data.
+In these cases, it could be useful to not include the data payload when logging.
+In some cases, an object id could be provided for more context, such as a Tenant/UserID instead
+of a username but only where the id belongs to a persisted object that an administrator can access.
+
+#### Avoid adding too much Context
```

Comment:
We should define the minimum necessary amount of information for context. In e.g. _"As a DataCater developer I will know where to look and debug in the codebase by reading the error message in the logs."_